src/_pytest/recwarn.py

```diff
@@ -2,10 +2,23 @@
 import inspect
 import re
 import warnings
+from types import TracebackType
+from typing import Any
+from typing import Callable
+from typing import Iterator
+from typing import List
+from typing import Optional
+from typing import overload
+from typing import Pattern
```

Comment:
if typing is imported anyway, why the if False below

Reply:
I answered this in the original PR but here's the longer version: https://github.com/asottile/flake8-typing-imports#rationale